### DISCUSSION TITLE
Logger Setup is not global

### DISCUSSION THREAD

**Discussion by jw2476** [denoland/deno#8723] [Q&A]
I define my desired logging setup eg:

```typescript
// main.ts

import * as log from "https://deno.land/std/log/mod.ts"

await log.setup({
    handlers: {
        console: new log.handlers.ConsoleHandler("DEBUG"),

        file: new log.handlers.FileHandler("WARNING", {
            filename: "./log.txt",
            formatter: "{levelName} {msg}",
        }),
    },
    loggers: {
        default: {
            level: "DEBUG",
            handlers: ["console", "file"],
        }
    }
});
```

Then when I use it in other files the configuration is not saved eg:

```typescript
// notMain.ts

import * as log from "https://deno.land/std/log/mod.ts"

log.debug("Hello, World!") // Nothing prints
```

I think this should be global as it will save a lot of code re-use

**Comment by kitsonk** [comment]
You probably want to consider exporting you configured log object from `main.ts` to ensure that the code runs in the correct order and that the logger is setup before you try to log, instead of importing it from the original module again.

**Reply by jw2476** [reply]
I cant define a logging config in my library because it's a library, it needs to be up to the user to decide. This is why I think it should be global, like Python and Java, because otherwise there won't be any logging in libraries...

**Reply by jw2476** [reply]
@kitsonk

**Reply by lucacasonato** [reply]
@jw2476 It is global. You need to initialize the logger _before_ importing any libraries.
